v1/services/retention/config.go

```diff
@@ -0,0 +1,49 @@
+package retention
+
+import (
+	"errors"
+	"time"
+
+	"github.com/influxdata/influxdb/v2/toml"
+	"github.com/influxdata/influxdb/v2/v1/monitor/diagnostics"
+)
+
+// Config represents the configuration for the retention service.
+type Config struct {
+	Enabled       bool          `toml:"enabled"`
+	CheckInterval toml.Duration `toml:"check-interval"`
+}
+
+// NewConfig returns an instance of Config with defaults.
+func NewConfig() Config {
+	return Config{Enabled: true, CheckInterval: toml.Duration(30 * time.Minute)}
+}
+
+// Validate returns an error if the Config is invalid.
+func (c Config) Validate() error {
+	if !c.Enabled {
+		return nil
+	}
+
+	// TODO: Should we enforce a minimum interval?
```

Comment:
I'm not sure it's worth adding minimums. Hopefully people won't be adjusting this config option at all.

Reply:
Will remove that comment